### PR TITLE
Dedupe on-prem bootstrap manifests

### PR DESCRIPTION
**Pull request by bcrochet**
**- What I did**
Currently, all of the on-prem platforms duplicate the manifests needed for
bootstrap. There is a small difference in all of them, and these can be
gotten rid of by moving some of the template code into functions in Go.

**- How to verify it**
Verify that on-prem platforms can still deploy.

**- Description for the changelog**

Dedupe bootstrap manifests for on-prem platforms.

### CODE REVIEW

**Review by cgwalters — APPROVED** [review]
/approve
the idea in general (and we should likely do the same with the kubelet systemd unit).

But this is a nontrivial change without an immediate bug it's fixing (right?) so I'm adding the 4.7 label.

**Comment by bcrochet** [comment]
> /approve
> the idea in general (and we should likely do the same with the kubelet systemd unit).
> 
> But this is a nontrivial change without an immediate bug it's fixing (right?) so I'm adding the 4.7 label.
> 

You are correct. Thanks!

File: pkg/operator/render.go [hunk 1]
```diff
@@ -156,3 +160,79 @@ func isSingleStackIPv6(serviceCIDRs []string) (bool, error) {
...
+		case configv1.BareMetalPlatformType:
+			return cfg.Infra.Status.PlatformStatus.BareMetal.NodeDNSIP, nil
+		case configv1.OvirtPlatformType:
+			return cfg.Infra.Status.PlatformStatus.Ovirt.NodeDNSIP, nil
+		case configv1.OpenStackPlatformType:
+			return cfg.Infra.Status.PlatformStatus.OpenStack.NodeDNSIP, nil
+		case configv1.VSpherePlatformType:
+			return cfg.Infra.Status.PlatformStatus.VSphere.NodeDNSIP, nil
+		default:
+			return nil, fmt.Errorf("invalid platform for API VIP")
```
Comment (cybertron) [inline pkg/operator/render.go:214]: s/API VIP/DNS IP/

File: pkg/operator/render.go [hunk 2]
```diff
@@ -156,3 +160,79 @@ func isSingleStackIPv6(serviceCIDRs []string) (bool, error) {
...
+		case configv1.BareMetalPlatformType:
+			return cfg.Infra.Status.PlatformStatus.BareMetal.IngressIP, nil
+		case configv1.OvirtPlatformType:
+			return cfg.Infra.Status.PlatformStatus.Ovirt.IngressIP, nil
+		case configv1.OpenStackPlatformType:
+			return cfg.Infra.Status.PlatformStatus.OpenStack.IngressIP, nil
+		case configv1.VSpherePlatformType:
+			return cfg.Infra.Status.PlatformStatus.VSphere.IngressIP, nil
+		default:
+			return nil, fmt.Errorf("invalid platform for API VIP")
```
Comment (cybertron) [inline pkg/operator/render.go:195]: s/API VIP/Ingress VIP/

File: pkg/operator/render.go [hunk 3]
```diff
@@ -156,3 +160,79 @@ func isSingleStackIPv6(serviceCIDRs []string) (bool, error) {
...
+		}
+	} else {
+		return nil, fmt.Errorf("")
+	}
+}
+
+func onPremPlatformNodeDNSIP(cfg mcfgv1.ControllerConfigSpec) (interface{}, error) {
```
Comment (cybertron) [inline pkg/operator/render.go:202]: Is this used? I'm not sure we need it now that we removed the DNS VIP.

**Review by cybertron — CHANGES_REQUESTED** [review]
Cool, this is a big improvement! A couple of things inline.

File: pkg/operator/render.go (hunk 3 above)
Comment (bcrochet) [reply pkg/operator/render.go:202]: Yeah. I haven't seen it in any manifests/templates. I'll remove it.

**Comment by ravidbro** [comment]
I'm thrilled to see that.
Are we planning to do the same also for the templates?

**Comment by bcrochet** [comment]
> I'm thrilled to see that.
> Are we planning to do the same also for the templates?

See #2079

**Review by cybertron — APPROVED** [review]
/lgtm

**Comment by cybertron** [comment]
/lgtm cancel

I don't think we want the bot to sit here and retest until 4.7 opens.

**Comment by cgwalters** [comment]
Thanks for pushing this through!

**Pull request merged**